dvc/output.py

```diff
@@ -887,7 +887,7 @@ def _set_obj_names(self, obj):
         obj.name = str(self)
         if isinstance(obj, Tree):
             for key, entry_obj in obj:
-                entry_obj.name = os.path.join(str(self), *key)
+                entry_obj.name = os.path.sep.join([obj.name, *key])
```

Comment:
strictly speaking, this should use `fs.sep` or `self.path_info.sep`(both are not available right now :slightly_frowning_face: )